File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    tangerine.sort()
    total = {}
    for key in tangerine:
        total[key] = total.get(key,0) +1
    
    total = sorted(total.items(), key=lambda x: x[1], reverse = True)
    
    for i in range(len(total)):
        if k<=0:
            return answer
        else:
            k-=total[i][1]
            answer+=1
    
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int k, vector<int> tangerine) {
    vector<int> cnt(10000001,0);
    for(int x:tangerine){
        cnt[x]++;
    }
    vector<int> frequency(tangerine.size()+1, 0);
    for(int x: cnt){
        if(x>0){
            frequency[x]++;
        }
    }
    int answer=0;
    for(int i = tangerine.size(); i>=1; i--){
        while (frequency[i]>0){
            k-=i;
            answer++;
            frequency[i]--;
            
            if(k<=0){
                return answer;
            }
        }
    }
    return answer;
}